RCTVideo.m

```diff
@@ -270,9 +270,19 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
           
         NSObject *width = @"undefined";
         NSObject *height = @"undefined";
+        NSString *orientation = @"undefined";
+        AVAssetTrack *vT = nil;
+
         if ([_playerItem.asset tracksWithMediaType:AVMediaTypeVideo].count > 0) {
-          width = [NSNumber numberWithFloat:[_playerItem.asset tracksWithMediaType:AVMediaTypeVideo][0].naturalSize.width];
-          height = [NSNumber numberWithFloat:[_playerItem.asset tracksWithMediaType:AVMediaTypeVideo][0].naturalSize.height];
+          vT = [[_playerItem.asset tracksWithMediaType:AVMediaTypeVideo] objectAtIndex:0];
+          width = [NSNumber numberWithFloat:vT.naturalSize.width];
+          height = [NSNumber numberWithFloat:vT.naturalSize.height];
+          CGAffineTransform txf = [vT preferredTransform];
+
+          if ((vT.naturalSize.width == txf.tx && vT.naturalSize.height == txf.ty) | (txf.tx == 0 && txf.ty == 0))
```

Comment:
Please use logical OR `||` instead of bitwise inclusive OR `|`.

http://www.dummies.com/how-to/content/objectivec-for-dummies-cheat-sheet.html